test/e2e/thanos_querier_test.go

```diff
@@ -91,3 +92,28 @@ func TestThanosQueryCanQueryWatchdogAlert(t *testing.T) {
 		},
 	)
 }
+
+func TestMonitoringApiReaderRole(t *testing.T) {
```

Comment:
I'd rather check that
2. a test service with the `cluster-monitoring-view` clusterrole has access to the API.
3. a test service with the `cluster-monitoring-api-reader` role has access to the API.
4. a test service account without any of these roles doesn't have access to the API